### DISCUSSION TITLE
add setting for cursor theme and size

### DISCUSSION THREAD

**Discussion by nnuel** [Manjaro-Sway/manjaro-sway#235] [Ideas]
I was struggling a bit after changing the cursor-theme, to set the right size. 
These were the two settings I had to put, one was through the command-line.

```
gsettings set org.gnome.desktop.interface cursor-size 40
```

and the other one in the `.config/sway/config`

```
seat * xcursor_theme Bibata_Spirit 40
```

Where Bibata_Spirit was the theme-name, that could be substituted for any. If I didn't set the gsettings, than the cursor would remain small on the waybar and in firefox and some other apps. 

Not sure how to trigger the gsettings command outside of a script or have the two combined, where one could set a single variable of $cursor-size = 40. Hope this is of any help for someone in the future.

**Comment by nnuel** [comment]
![image](https://user-images.githubusercontent.com/4148154/150825166-d1b8278f-082c-4c62-883b-e1d230bf44f1.png)

I found this in:

```
/etc/sway/config.d/90-enable-gtk-theme.conf
```

so I suggest an extra line:

`gsettings set org.gnome.desktop.interface cursor-size $cursor-size`

and in sway config:

`seat * xcursor_theme $cursor-theme $cursor-size`

That should do it I guess, to have it working in all the programs and allows for cursor-theme changes directly in the config.

One note: This requires a cursor-theme with built-in multiple sizes. I can foresee it not working, if the default theme only comes in one size

**Reply by boredland** [reply]
I guess we could add a variables for cursor theme and size to the config and run the gesettings cmd on sway load. have you tried this with a kde application?

> One note: This requires a cursor-theme with built-in multiple sizes. I can foresee it not working, if the default theme only comes in one size

Good point. Should we probably switch to another theme then? Recommendations?

**Comment by boredland** [comment]
This is part of the theme now!
